### PR TITLE
Generate a best-effort SSH-based push URL for component repos

### PR DESCRIPTION
**Pull request by simu**
This change assumes that most Git hosting services will use the pattern `ssh://git@example.com/path/to/repo.git` for SSH-based push URLs.

We configure each component repo's push-URL using the assumed substitution. This may not always generate a working push-URL, but for popular services like GitHub and GitLab, this removes the need of manually fixing the origin remote when trying to push a feature branch.

## Checklist

- [x] Keep pull requests small so they can be easily reviewed.
- [x] Update the documentation.
- [x] Update tests.
- [x] Update the ./CHANGELOG.md.

### CODE REVIEW

**Review by srueg — APPROVED** [review]

File: tests/test_component.py [hunk 1]
```diff
@@ -43,12 +46,52 @@ def _setup_component(
...
+    assert push_remote == repo_url
+
+
+@pytest.mark.parametrize(
+    "repo_url,push_url",
+    [
+        ("http://host/path/to/repo.git", "ssh://git@host/path/to/repo.git"),
+        ("https://host/path/to/repo.git", "ssh://git@host/path/to/repo.git"),
+        ("https://user@host/path/to/repo.git", "ssh://git@host/path/to/repo.git"),
+        ("https://user:pass@host/path/to/repo.git", "ssh://git@host/path/to/repo.git"),
```
Comment (corvus-ch) [inline tests/test_component.py:72]: What happens to `http://example.com:1234/path/to/repo.git` or http://user:pass@example.com:1234/path/to/repo.git?

**Review by corvus-ch — CHANGES_REQUESTED** [review]
There is an edge case not covered by the tests. Please update.

File: tests/test_component.py (hunk 1 above)
Comment (simu) [reply tests/test_component.py:73]: I've refactored the implementation to drop everything but the host and path of the input URL, if the url starts with `http://` or `https://`.

Obviously we can't guess a potential non-standard SSH port for components which are served on a non-standard HTTP(S) port.

**Review by corvus-ch — APPROVED** [review]
